mcf-browser/src/back-end/data.ts

```diff
@@ -0,0 +1,250 @@
+   * and instantiates all of the properties
+   * @param {DataPoint[]} data the datapoints for the series
+   * @param {string} variableMeasured the variableMeasured of a series
+   * @param {string} observationAbout the observationAbout of a series
+   * @param {string} provenance the provenance of a series
+   * @param {string} measurementMethod the measurementMethod of a series
+   * @param {string} observationPeriod the observationPeriod of a series
+   * @param {string} unit the unit of a series
+   * @param {number} scalingFactor the scalingFactor of a series
+   */
+  constructor(
+      data: DataPoint[],
+      variableMeasured?: string,
+      observationAbout?: string,
+      provenance?: string,
+      measurementMethod?: string,
+      observationPeriod?: string,
+      unit?: string,
+      scalingFactor?: number,
+  ) {
+    const cleanVariable = (variable: string | undefined) => {
+      if (!variable) {
+        return variable;
+      } else if (variable.startsWith('dcs:')) {
+        return variable.slice(4);
+      } else if (variable.startsWith('dcid:')) {
+        return variable.slice(5);
+      } else if (variable.startsWith('schema:')) {
+        return variable.slice(7);
+      }
+      return variable;
+    };
+
+    this.data = data;
+    this.variableMeasured = cleanVariable(variableMeasured);
+    this.observationAbout = cleanVariable(observationAbout);
+    this.provenance = cleanVariable(provenance);
+    this.measurementMethod = cleanVariable(measurementMethod);
+    this.observationPeriod = cleanVariable(observationPeriod);
+    this.unit = cleanVariable(unit);
+    this.scalingFactor = scalingFactor;
+
+    this.id = Series.toID(
+        variableMeasured,
+        observationAbout,
+        provenance,
+        measurementMethod,
+        observationPeriod,
+        unit,
+        scalingFactor,
+    );
+  }
+
+  /**
+   * Generates a unique ID for a series given the values of
+   * the properties of a Series object
+   * @param {string} variableMeasured the variableMeasured of a series
+   * @param {string} observationAbout the observationAbout of a series
+   * @param {string} provenance the provenance of a series
+   * @param {string} measurementMethod the measurementMethod of a series
+   * @param {string} observationPeriod the observationPeriod of a series
+   * @param {string} unit the unit of a series
+   * @param {number} scalingFactor the scalingFactor of a series
+   * @return {string} the id
+   */
+  static toID(
+      variableMeasured?: string,
+      observationAbout?: string,
+      provenance?: string,
+      measurementMethod?: string,
+      observationPeriod?: string,
+      unit?: string,
+      scalingFactor?: number,
+  ) : string {
+    const facetList = [
+      variableMeasured ? variableMeasured : '',
+      observationAbout ? observationAbout : '',
+      provenance ? provenance : '',
+      measurementMethod ? measurementMethod : '',
```

Comment:
These must be set?

Reply:
You're right, most of them don't need to be set except for scalingFactor which is a number, it's been updated!